akka-http-core/src/main/scala/akka/http/engine/parsing/BodyPartParser.scala

```diff
@@ -115,9 +116,8 @@ private[http] final class BodyPartParser(defaultContentType: ContentType,
       }
       rec(offset)
     } catch {
-      case NotEnoughDataException ⇒ continue(input.takeRight(needle.length), 0)(parsePreamble)
+      case NotEnoughDataException ⇒ continue(input.takeRight(needle.length + 2), 0)(parsePreamble)
```

Comment:
These lines are only executed on "parsing boundaries" where an intermediate chunk of data coming in from the TCP layer ends. So, things are actually somewhat self-adjusting. If the system becomes more loaded the chunks become bigger (as reading from the socket buffer become less frequent).

Also, it is doubtful that *always* allocating an extra function instance even if the preamble fits in a single chunk (or isn't even present, as in the majority of cases) vs. allocating one on every chunk boundary does indeed provide any benefit.